src/object/schema_system_catalog_builder.cpp

```diff
@@ -111,26 +129,36 @@ namespace cubschema
       {
 	DB_CONSTRAINT_TYPE type = c.type;
 
-	const char *name = c.name.data();
+	const char *name = nullptr;
+	if (!c.name.empty ())
+	  {
+	    name = c.name.data();
+	  }
+
 	int is_class_attribute = c.is_class_attributes ? 1 : 0;
-	if (type == DB_CONSTRAINT_INDEX)
+
+	switch (type)
+	  {
+	  case DB_CONSTRAINT_INDEX:
+	  case DB_CONSTRAINT_PRIMARY_KEY:
+	  case DB_CONSTRAINT_UNIQUE:
+	  case DB_CONSTRAINT_NOT_NULL:
```

Comment:
Are the following three cases not covered here?
- DB_CONSTRAINT_REVERSE_UNIQUE
- DB_CONSTRAINT_REVERSE_INDEX
- DB_CONSTRAINT_FOREIGN_KEY

Reply:
Yes. If these cases are used in the future, they should be added. Since they have never been used until now, I didn't implement them.